### PR TITLE
Trac 42022: Customize: Fix date time control defects

### PR DESCRIPTION
**Pull request by westonruter**
https://core.trac.wordpress.org/ticket/42022

### CODE REVIEW

File: src/wp-admin/css/customize-controls.css [hunk 1]
```diff
@@ -268,18 +250,38 @@ body.outer-section-open .wp-full-overlay.expanded .wp-full-overlay-main {
 	padding-bottom:10px;
 }
 
-.date-time-fields label,
+.date-time-fields .date-input,
 .date-time-fields .date-time-separator {
 	float: left;
-	margin-right:5px;
+	margin-right: 5px !important;
```
Comment (sayedtaqui) [inline src/wp-admin/css/customize-controls.css]: I think here we can increase the priority by being a bit more specific `.customize-control.customize-control-date_time .date-time-fields .date-input`, `!important` would probably not be required.

File: src/wp-admin/js/customize-controls.js [hunk 2]
```diff
@@ -4156,35 +4152,51 @@
...
 				element = new api.Element( input );
-				element.validate = function( value ) {
-					return _.contains( [ 'am', 'pm' ], value ) ? value : parseInt( value, 10 );
-				};
+				if ( 'meridian' === component ) {
+					element.validate = function( value ) {
+						if ( 'am' !== value && 'pm' !== value ) {
+							return null;
+						}
+						return value;
```
Comment (sayedtaqui) [inline src/wp-admin/js/customize-controls.js]: or may be in one line do `return ( 'am' !== value && 'pm' !== value ) ? null : value`

File: src/wp-includes/customize/class-wp-customize-date-time-control.php [hunk 3]
```diff
@@ -101,68 +101,56 @@ public function content_template() {
 		?>
 
 		<# _.defaults( data, <?php echo wp_json_encode( $data ); ?> ); #>
+		<# var idPrefix = 'el' + String( Math.random() ); #>
```
Comment (sayedtaqui) [inline src/wp-includes/customize/class-wp-customize-date-time-control.php]: This creates a very ugly looking id like `el0.9791353552326172`  , can we use `_.uniqueId( data.type )`  instead?

File: src/wp-includes/customize/class-wp-customize-date-time-control.php [hunk 4]
```diff
@@ -101,68 +101,56 @@ public function content_template() {
...
-						<span class="screen-reader-text"><?php esc_html_e( 'Minute' ); ?></span>
-						<input type="number" size="2" maxlength="2" autocomplete="off" class="date-input minute" data-component="minute" min="0" max="59" />
-					</label>
+					<label for="{{ idPrefix }}date-time-minute" class="screen-reader-text"><?php esc_html_e( 'Minute' ); ?></label>
+					<input id="{{ idPrefix }}date-time-minute" type="number" size="2" autocomplete="off" class="date-input minute" data-component="minute" min="0" max="59">
 					<# if ( data.twelveHourFormat ) { #>
-					<label class="am-pm-field">
-						<span class="screen-reader-text"><?php esc_html_e( 'AM / PM' ); ?></span>
-						<select class="date-input" data-component="ampm">
+						<label for="{{ idPrefix }}date-time-meridian" class="screen-reader-text"><?php esc_html_e( 'Meridian' ); ?></label>
```
Comment (sayedtaqui) [inline src/wp-includes/customize/class-wp-customize-date-time-control.php]: Yaass! "Meridian" was the word 👍  ,  AM/PM was looking weird.

File: src/wp-admin/js/customize-controls.js (hunk 2 above)
Comment (westonruter) [reply src/wp-admin/js/customize-controls.js]: The `if` statement seems easier for me to read.

**Comment by westonruter** [comment]
@sayedtaqui I've updated the date/time demo plugin to make use of this new `'include_time' => false` param: https://gist.github.com/westonruter/ca1a77f9309af4ab258573cee5ed9d30

**Comment by westonruter** [comment]
Committed to `trunk` in r41670.

**Pull request closed without merging**